### ISSUE TITLE
Fitting intercept differs between sparse and dense implementations

### ISSUE THREAD

**Issue by jolars** [jolars/sgdnet#2]
The source in scikit-learn uses `intercept_decay` to shrink the learning rate when features are sparse in order to prevent intercept oscillation. There is additional information at <https://stackoverflow.com/questions/46835181/sklearn-perceptron-behaving-differently-for-sparse-and-dense>. 

The problem is that the output differs depending on whether features are sparse or not, which is not what one expects from the function.

Possible solutions:

* Center elements in-place as glmnet does.
* Keep it as it is. Perhaps throwing a warning when the intercept is fit and features are sparse.
* Make the implementations equivalent and accept the possibility of convergence issues.
* Don't allow the intercept to be fit when features are sparse, i.e. throw an error.
